pkg/api/handler.go

```diff
@@ -173,3 +176,41 @@ func (h *handler) resetPassword(c echo.Context) error {
 		"success": true,
 	})
 }
+
+func (h *handler) signinWithProvider(c echo.Context) error {
+	payload := new(oauth.Oauth2Payload)
+
+	if err := c.Bind(payload); err != nil {
+		return echo.NewHTTPError(http.StatusInternalServerError, "An error occurred with your payload")
+	}
+	provider := c.Param("provider")
+	if provider == "" {
```

Comment:
I think that this test is useless as if the provider is empty, you'll go the the `default` case of the switch.